portable/GCC/ARM_CR5/portASM.S

```diff
@@ -45,7 +45,10 @@
 	.extern vTaskSwitchContext
 	.extern vApplicationIRQHandler
 	.extern ulPortInterruptNesting
+
+#if( configUSE_TASK_FPU_SUPPORT != 0 )

```

Comment:
Good point. I was having trouble with non-assembler compatible definitions in my FreeRTOSConfig. I like @ChristosZosi 's approach using the `__ARM_FP` macro instead, so I'll cherry-pick those commits over here.

Given that libc might use FP registers for things like memcpy, I strongly prefer always having an FP context for each task when the compile has FP enabled rather than configUSE_TASK_FPU_SUPPORT.

Reply:
I like that approach too especially the part catches the incorrect configUSE_TASK_FPU_SUPPORT configurations - 
```
#if configUSE_TASK_FPU_SUPPORT == 0
    #ifdef __ARM_FP
        #error The Floating Point Unit (FPU) of the ARM Cortex-R5 is enabled. Therefore you must   \
               define configUSE_TASK_FPU_SUPPORT either to 1 or 2.
    #endif /* __ARM_FP*/
#elif (configUSE_TASK_FPU_SUPPORT == 1) || (configUSE_TASK_FPU_SUPPORT == 2)
    #ifndef __ARM_FP
        #error The Floating Point Unit (FPU) of the ARM Cortex-R5 is disabled. Therefore you must \
               define configUSE_TASK_FPU_SUPPORT to 0.
    #endif /* __ARM_FP*/
#endif /* configUSE_TASK_FPU_SUPPORT */
```

Reply:
Hit @paulbartell ,

I am not sure what the typical workflow is, but I could push my changes on top of yours to your PR branch if this is desired

Reply:
I rebased this PR using your branch.